qmf/src/plugins/messageservices/imap/imapstrategy.cpp

```diff
@@ -3613,11 +3647,6 @@ void ImapRetrieveMessageListStrategy::folderListFolderAction(ImapStrategyContext
     uint minimum(_minimum);
     QMailMessageKey sourceKey(QMailDisconnected::sourceKey(properties.id));
 
-    // Purge messages marked as removed, facilitates detection of messages missign on client, and prevents cruft building up
-    if (!purge(context, sourceKey & QMailMessageKey::status(QMailMessage::Removed))) {
-        _error = true;
-    }
-
     // Could get flag changes mod sequences when CONDSTORE is available
```

Comment:
We should not purge here otherwise messages marked as deleted keep being downloaded. Test with servers using CONDSTORE/QRESYNC and without that purge for messages that are really gone is correctly handled already at the end of the strategies.